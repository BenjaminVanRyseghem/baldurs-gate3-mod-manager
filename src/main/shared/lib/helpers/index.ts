export { appPaths } from "./appPaths";

export { ipcRequestHandler } from "./ipcRequestHandler";

export { resolveHtmlPath } from "./resolveHtmlPath";

export { t } from "./trpc";
