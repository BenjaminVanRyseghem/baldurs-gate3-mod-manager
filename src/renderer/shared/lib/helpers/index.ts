export { classNames } from "./classNames";

export { compose } from "./compose";

export { delay } from "./delay";

export { lazyImport } from "./lazyImport";

export { mergeRefs } from "./mergeRefs";

export { createTrpcClient, trpc } from "./trpc";
