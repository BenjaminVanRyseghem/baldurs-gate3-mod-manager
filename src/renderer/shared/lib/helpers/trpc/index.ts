export { trpc } from "./trpc";

export { createTrpcClient } from "./createTrpcClient";
