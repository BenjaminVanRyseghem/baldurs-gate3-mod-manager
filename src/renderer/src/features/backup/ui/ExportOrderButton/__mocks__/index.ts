export { SaveSettingsButtonMock as SaveSettingsButton } from "./ExportOrderButton.mock";
