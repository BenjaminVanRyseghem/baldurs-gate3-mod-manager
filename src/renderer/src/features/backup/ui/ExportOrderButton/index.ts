export { ExportOrderButton } from "./ExportOrderButton";

export * from "./ExportOrderButton.type";
