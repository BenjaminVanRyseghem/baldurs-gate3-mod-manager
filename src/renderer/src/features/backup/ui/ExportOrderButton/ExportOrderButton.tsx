import { delay } from "@common/lib";
import { classNames, trpc } from "@renderer/shared/lib/helpers";
import { Button } from "@renderer/shared/ui";

import css from "./ExportOrderButton.module.scss";
import { ExportOrderButtonProps } from "./ExportOrderButton.type";

const ExportOrderButton = ({ className, game }: ExportOrderButtonProps) => {
  const utils = trpc.useUtils();

  const { mutate, isLoading } = trpc.backup.exportBG3MMOrder.useMutation({
    onMutate: async (variables) => {
      await delay(850);

      return variables;
    },
    onSettled: () => {
      utils.backup.invalidate();
    },
  });

  return (
    <Button
      className={classNames(css.ExportOrderButton, className)}
      data-testid="ExportOrderButton"
      onClick={() => mutate(game)}
      loading={isLoading}
      disabled={isLoading}
    >
      Export Order to BG3MM format
    </Button>
  );
};

export { ExportOrderButton };
