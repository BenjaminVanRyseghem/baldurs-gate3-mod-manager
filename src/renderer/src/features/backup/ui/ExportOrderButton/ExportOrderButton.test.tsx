import { render } from "@testing-library/react";

import { withQueryProvider } from "@renderer/shared/lib/helpers/testUtils";

import { ExportOrderButton } from "./ExportOrderButton";

describe("<SaveSettingsButton />", () => {
  const game = "BG3";
  const Component = <ExportOrderButton game={game} />;

  it("should be defined", () => {
    expect(ExportOrderButton).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(withQueryProvider(Component));

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(withQueryProvider(Component));
    const element = getByTestId("SaveSettingsButton");

    expect(element).toBeDefined();
  });
});
