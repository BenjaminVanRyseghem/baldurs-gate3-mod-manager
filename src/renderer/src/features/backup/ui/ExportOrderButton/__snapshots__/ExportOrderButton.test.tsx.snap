// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ExportOrderButton /> > should match the snapshot 1`] = `
<div>
  <button
    class="Button ExportOrderButton contained primary"
    data-testid="ExportOrderButton"
    type="button"
  >
    <svg
      class="Spinner spinner"
    >
      bars.svg
    </svg>
    <span
      class="content"
    >
      Backup Current Settings
    </span>
  </button>
</div>
`;
